contracts/contracts/Job.sol

```diff
@@ -0,0 +1,360 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+contract Job {
+
+    struct JobData {
+        address supplier;
+        uint bounty;
+        address engineer;
+        uint deposit;
+
+        uint startTime;
+        uint completedTime;
+
+        bool closedBySupplier;
+        bool closedByEngineer;
+
+        States state;
+    }
+
+    uint public jobCount;
+    mapping(uint => JobData) public jobs;
+
+    address public owner;
+    address public paymentToken;
```

Comment:
So will any ERC20 token work? Like would we deploy a new contract for, say, DAI?

Reply:
That's exactly right.  AlthoughWe could deploy 2 different contracts - one for each payment token type (USDC, DAI, etc.).  Any ERC20 token will work.